eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenDualFunded.scala

```diff
@@ -435,43 +439,50 @@ trait ChannelOpenDualFunded extends DualFundingHandlers with ErrorHandlers {
-            remoteNodeId, nodeParams, fundingParams,
-            d.localPushAmount, d.remotePushAmount,
-            d.metaCommitments.params,
-            d.commitments.localCommit.spec.commitTxFeerate,
-            d.commitments.remoteCommit.remotePerCommitmentPoint,
-            wallet))
-          txBuilder ! InteractiveTxBuilder.Start(self, d.allFundingTxs)
-          stay() using d.copy(rbfStatus = RbfStatus.RbfInProgress(txBuilder)) sending TxAckRbf(d.channelId, fundingParams.localAmount)
+        d.rbfStatus match {
+          case RbfStatus.NoRbf =>
+            if (msg.feerate < minNextFeerate) {
+              log.info("rejecting rbf attempt: the new feerate must be at least {} (proposed={})", minNextFeerate, msg.feerate)
+              stay() using d.copy(rbfStatus = RbfStatus.RbfAborted) sending TxAbort(d.channelId, InvalidRbfFeerate(d.channelId, msg.feerate, minNextFeerate).getMessage)
+            } else if (d.remotePushAmount > msg.fundingContribution) {
+              log.info("rejecting rbf attempt: invalid amount pushed (fundingAmount={}, pushAmount={})", msg.fundingContribution, d.remotePushAmount)
+              stay() using d.copy(rbfStatus = RbfStatus.RbfAborted) sending TxAbort(d.channelId, InvalidPushAmount(d.channelId, d.remotePushAmount, msg.fundingContribution.toMilliSatoshi).getMessage)
+            } else {
+              log.info("our peer wants to raise the feerate of the funding transaction (previous={} target={})", d.fundingParams.targetFeerate, msg.feerate)
+              val fundingParams = InteractiveTxParams(
+                d.channelId,
+                d.fundingParams.isInitiator,
+                d.fundingParams.localAmount, // we don't change our funding contribution
+                msg.fundingContribution,
+                d.fundingParams.fundingPubkeyScript,
+                msg.lockTime,
+                d.fundingParams.dustLimit,
+                msg.feerate,
+                d.fundingParams.requireConfirmedInputs,
+              )
+              val txBuilder = context.spawnAnonymous(InteractiveTxBuilder(
+                remoteNodeId, nodeParams, fundingParams,
+                d.localPushAmount, d.remotePushAmount,
+                d.metaCommitments.params,
+                d.commitments.localCommit.spec.commitTxFeerate,
+                d.commitments.remoteCommit.remotePerCommitmentPoint,
+                d.commitments.remoteNextCommitInfo.toOption.get,
+                wallet))
+              txBuilder ! InteractiveTxBuilder.Start(self, d.allFundingTxs)
+              stay() using d.copy(rbfStatus = RbfStatus.RbfInProgress(txBuilder)) sending TxAckRbf(d.channelId, fundingParams.localAmount)
+            }
+          case RbfStatus.RbfAborted =>
+            log.info("rejecting rbf attempt: our previous tx_abort was not acked")
+            stay() sending Warning(d.channelId, InvalidRbfTxAbortNotAcked(d.channelId).getMessage)
+          case _ =>
```

Comment:
```suggestion
          case _: RbfStatus.RbfRequested | _: RbfStatus.RbfInProgress =>
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2536/commits/bbd9610daba263d41fc16b1f4734c56506e228e5